### PR TITLE
[MERGE] Fix problems with the forecasting methods and update unit tests. Some new techniques and parameters were added in the process.

### PR DESCRIPTION
**Pull request by zaoral**
Task [vx#3989](https://www.vauxoo.com/web#id=3989&view_type=form&model=project.task&action=138) HU [913](https://www.vauxoo.com/web#id=913&view_type=form&model=user.story&action=726) Sprint 18

### CODE REVIEW

File: stock_forecasting_smoothing_techniques/models/forecasting_smoothing_techniques.py [hunk 1]
```diff
@@ -12,6 +12,9 @@
 
 from openerp import models, fields, api, _
 from openerp.exceptions import ValidationError
+# import logging
+
+# _logger = logging.getLogger(__name__)
```
Comment (nhomar) [inline stock_forecasting_smoothing_techniques/models/forecasting_smoothing_techniques.py]: no commented lines if not used.

File: stock_forecasting_smoothing_techniques/models/forecasting_smoothing_techniques.py [hunk 2]
```diff
@@ -224,26 +263,30 @@ def _compute_weighted_move_average(self):
...
-        ma_error = 0.0
-        weight = (period * (period + 1)) / 2
-        for item in range(0, len(fv_list)):
-            fv_set = fv_list[item:item+period]
+        fperiod = float(period)
+        avg = [None for item in range(4)]
+        ma_error = []
+        weight = (fperiod * (fperiod + 1.0)) / 2.0
+        for item in range(period, numv+1):
+            #fv_set = fv_list[item-period+1:item+1]
```
Comment (nhomar) [inline stock_forecasting_smoothing_techniques/models/forecasting_smoothing_techniques.py]: Commented lines.

File: stock_forecasting_smoothing_techniques/models/forecasting_smoothing_techniques.py [hunk 3]
```diff
@@ -224,26 +263,30 @@ def _compute_weighted_move_average(self):
...
-                 for value in fv_set])]
-            ma_error += abs(avg[-1] - fv_set[-1])
+                [((day)/ weight) * value
+                 for (day, value) in enumerate(fv_set, 1)])]
+            ma_error += [abs(avg[-1] - fv_set[-1])]
+            # _logger.debug(
+            #     '{num:02d} avg {avg:10f} mae {mae:10f} weight {weight} set {fset} '.format(
+            #         num=item, fset=fv_set, avg=avg[-1], mae=ma_error[-1],
+            #         weight=weight,
+            #         ))
```
Comment (nhomar) [inline stock_forecasting_smoothing_techniques/models/forecasting_smoothing_techniques.py]: commented

**Comment by nhomar** [comment]
Kathy.

Please add thest cases when elements are empty please in order to check cases where system receive null values.

**Comment by zaoral** [comment]
ok. start working on it.

**Comment by zaoral** [comment]
@nhomar this branch is ready to review

Already apply the changes you request (remove comments, add crud test, fix pylint / flake8).
Runbot return green as you can see in this [build](http://runbot.vauxoo.com/runbot/team/yoytec-24?testing=1&running=1&done=1&limit=10000&pending=1#119pull_4)

Best Regards.

**Pull request merged**